### PR TITLE
#3602 Fix the way next.config.js file is created by the codemod

### PR DESCRIPTION
**Pull request by siddhsuresh**
Closes: #3602

- [x] extract config from existing blitz.config.(ts|js)
- [x] place config wrapped with ```withBlitz``` funtion in the next.config.js file

I did not rename the file from blitz.config.(ts|js) to next.config.js or delete the blitz config file as it was necessary to run other steps in the codemod
# Examples
[Using the examples in the legacy blitz repository to test the codemod](https://github.com/blitz-js/legacy-framework/tree/main/exampleshttps://github.com/blitz-js/legacy-framework/tree/main/examples)
### Before: 
[blitz.config.js](https://github.com/blitz-js/legacy-framework/blob/main/examples/auth/blitz.config.ts)
```js
module.exports = {
  middleware: [
    (req, res, next) => {
      res.blitzCtx.referer = req.headers.referer
      return next()
    },
  ],
  log: {
    // level: "trace",
  },
  experimental: {
    isomorphicResolverImports: true,
  },
  // webpack: (config, { buildId, dev, isServer, defaultLoaders, webpack }) => {
  //   // Note: we provide webpack above so you should not `require` it
  //   // Perform customizations to webpack config
  //   // Important: return the modified config
  //   return config
  // },
  // webpackDevMiddleware: (config) => {
  //   // Perform customizations to webpack dev middleware config
  //   // Important: return the modified config
  //   return config
  // },
}
```
### After:
next.config.js
```js
import { withBlitz } from "@blitzjs/next";

const config = withBlitz({
   middleware: [
    (req, res, next) => {
      res.blitzCtx.referer = req.headers.referer
      return next()
    },
  ],
  log: {
    // level: "trace",
  },

  // webpack: (config, { buildId, dev, isServer, defaultLoaders, webpack }) => {
  //   // Note: we provide webpack above so you should not `require` it
  //   // Perform customizations to webpack config
  //   // Important: return the modified config
  //   return config
  // },
  // webpackDevMiddleware: (config) => {
  //   // Perform customizations to webpack dev middleware config
  //   // Important: return the modified config
  //   return config
  // },
  experimental: {
    isomorphicResolverImports: true,
  }
});

module.exports = config;
```

### Before:
[blitz.config.ts](https://github.com/blitz-js/legacy-framework/blob/main/examples/store/blitz.config.js)
```ts
import {sessionMiddleware, simpleRolesIsAuthorized} from "blitz"
import db from "db"
const withBundleAnalyzer = require("@next/bundle-analyzer")({
  enabled: process.env.ANALYZE === "true",
})

module.exports = withBundleAnalyzer({
  middleware: [
    sessionMiddleware({
      cookiePrefix: "blitz-auth-example",
      isAuthorized: simpleRolesIsAuthorized,
      // sessionExpiryMinutes: 4,
      getSession: (handle) => db.session.findFirst({where: {handle}}),
    }),
  ],
  cli: {
    clearConsoleOnBlitzDev: false,
  },
  codegen: {
    templateDir: "./my-templates",
  },
  log: {
    // level: "trace",
  },
  experimental: {
    initServer() {
      console.log("Hello world from initServer")
    },
  },
  /*
  webpack: (config, {buildId, dev, isServer, defaultLoaders, webpack}) => {
    // Note: we provide webpack above so you should not `require` it
    // Perform customizations to webpack config
    // Important: return the modified config
    return config
  },
  webpackDevMiddleware: (config) => {
    // Perform customizations to webpack dev middleware config
    // Important: return the modified config
    return config
  },
  */
})
```
### After:
next.config.js
```js
import { withBlitz } from "@blitzjs/next";
import {sessionMiddleware, simpleRolesIsAuthorized} from "blitz"
import db from "db"
const withBundleAnalyzer = require("@next/bundle-analyzer")({
  enabled: process.env.ANALYZE === "true",
})

const config = withBlitz(withBundleAnalyzer({
  middleware: [
    sessionMiddleware({
      cookiePrefix: "blitz-auth-example",
      isAuthorized: simpleRolesIsAuthorized,
      // sessionExpiryMinutes: 4,
      getSession: (handle) => db.session.findFirst({where: {handle}}),
    }),
  ],
cli: {
    clearConsoleOnBlitzDev: false,
  },

  codegen: {
    templateDir: "./my-templates",
  },

  log: {
    // level: "trace",
  },

  experimental: {
    initServer() {
      console.log("Hello world from initServer")
    },
  }
  /*
  webpack: (config, {buildId, dev, isServer, defaultLoaders, webpack}) => {
    // Note: we provide webpack above so you should not `require` it
    // Perform customizations to webpack config
    // Important: return the modified config
    return config
  },
  webpackDevMiddleware: (config) => {
    // Perform customizations to webpack dev middleware config
    // Important: return the modified config
    return config
  },
  */
}));

module.exports = config;
```

### CODE REVIEW

**Review by dillonraphael — COMMENTED** [review]
Can we remove all the comments?

**Comment by siddhsuresh** [comment]
okay I shall add a commit without the comments

**Comment by dillonraphael** [comment]
@siddhsuresh Ah we merged some PR to main & now there is conflicts. I can try and fix later if you don't have time.

**Comment by siddhsuresh** [comment]
@dillonraphael, I think I resolved the merge conflict

**Review by dillonraphael — COMMENTED** [review]
Just tested your branch. Works great.

But I think we have an issue with our flow.

Would it be possible to remove this block of code? Then I'll create another PR to handle middlewares in the step `Move middleware to blitz server file`

File: packages/codemod/src/upgrade-legacy.ts [hunk 1]
```diff
@@ -49,37 +49,58 @@ const upgradeLegacy = async () => {
...
+            createdConfig.arguments[0].properties = createdConfig.arguments[0].properties.filter(
+              (node) => node.key.name !== "middleware",
+            )
+          else {
+            throw new ExpectedError(
+              "Middleware has not been removed from next.config.js. It has to be removed manually",
+            )
+          }
+        }
+      }
```
Comment (dillonraphael) [inline packages/codemod/src/upgrade-legacy.ts:85]: I think we should remove this. Then in the `Move middleware to blitz server file` step we can handle the middlewares.

**Review by dillonraphael — CHANGES_REQUESTED** [review]

File: packages/codemod/src/upgrade-legacy.ts (hunk 1 above)
Comment (dillonraphael) [reply packages/codemod/src/upgrade-legacy.ts:85]: Would you be able to remove this block? Then I'll merge the PR and then I'll create a new one fixing the middlewares in the correct step.

Comment (siddhsuresh) [reply packages/codemod/src/upgrade-legacy.ts]: @dillonraphael I have now committed the requested change

**Pull request merged**